packages/mui-material/src/StepLabel/StepLabel.d.ts

```diff
@@ -52,7 +59,7 @@ export interface StepLabelProps extends StandardProps<React.HTMLAttributes<HTMLD
   /**
    * The component to render in place of the [`StepIcon`](/material-ui/api/step-icon/).
    */
-  StepIconComponent?: React.ElementType;
+  StepIconComponent?: React.ElementType<StepLabelIconComponentProps>;
```

Comment:
It should use the `StepIconProps`:
 ```suggestion
   StepIconComponent?: React.ElementType<StepIconProps>;
```